src/main/java/com/programmerdan/minecraft/simpleadminhacks/autoload/AutoLoad.java

```diff
@@ -27,5 +27,12 @@
 	 * @return Whether this option is required
 	 */
 	boolean isRequired() default true;
+
+	/**
+	 * The type of parser used to process the data retrieved from the config.
+	 *
+	 * @return Returns what processor to use for retrieved config data.
+	 */
+	DataParser processor() default DataParser.DEFAULT;
```

Comment:
It'd be cool to auto detect this. Have a Class --> DataParser static mapping in the enum and use it to retrieve the proper DataProcessor. For lists just cast the list, retrieve the first element and do a lookup for the class of that object (and maybe recursively for the super class)